Feat/page profile

manvinderjit opened the pull request:
## Summary

Add a profile page to show the profile for an individual volunteer.

## Additional information

The profile page right now uses a profile Id _**'pid'**_  URL parameter to fetch the profile id. Let me know if this has to be updated.

manvinderjit committed:
Rename: Rename Profiles Page

Rename profile/* to profiles/* as it shows multiple files.
Update <a> in index menu.

manvinderjit committed:
feat: Add Profile page

Add Profile page for showing volunteer's profile details.

manvinderjit committed:
feat: Profile Page

Add CSS styles for Profile page.
Add SVG icons as React components.
Add links for profiles on index page.

manvinderjit commented:
@madcampos This page displays profile for a single individual. @kbventures you have already written a _<ProfileList>_ component for rendering all profiles as a list. Is it sufficient for the use case and are you planning to work on it further?

madcampos commented on src/components/ProfileCard/ProfileCard.tsx:
```diff
@@ -0,0 +1,155 @@
+					setIsErrorProfile(true);
+				}
+				const jsonDataProfile = await responseProfile.json();
+				// REMOVE: Filtering won't be required with API, remove filtering when integrating with backend
```
**suggestion**: Use [common pattern words](https://stackoverflow.com/a/9586701/937851) like "TODO" or "FIXME".
Some extensions also help to highlight different keywords with different colors. Here is the one I use: https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments

kbventures replied:
@manvinderjit we should definitely reuse that if possible.  I'm not currently working on it but we shall surely expand on this in terms of functionality.

kbventures replied:
@manvinderjit for front end work we should try to include a picture so we can get a quick peak without running the code(even though we will run it).  Eventually when we have mobile version both images will be expected.

manvinderjit replied:
![PreviewInBrowser](https://github.com/user-attachments/assets/50f5c2d7-c06f-4ecc-b004-5b4ad0b09b06)
@kbventures  As each commit creates and updates a preview branch in CloudFlare, we can preview it directly in the browser instead of running locally. Refer to the image for instructions. Apologies for it being a crude design.
If we still need images, I can upload them.

madcampos commented:
@manvinderjit
> This page displays profile for a single individual.
Not sure if it was a question, but yes, it is correct. And the layout look nice 😉

> you have already written a _ProfileList_ component for rendering all profiles as a list. Is it sufficient for the use case and are you planning to work on it further?

I think for now this is sufficient, we can get back to it later once we have the API for actually querying the data.

madcampos commented:
Please test the page for mobile using the dev tools to check for potential issues with content breaking.

![localhost_3000_pages_profile__pid=2](https://github.com/user-attachments/assets/2dcbda94-1947-4806-a751-91d797594bde)

kbventures commented:
@madcampos Checking for mobile may become more stringent once we receive clear guidelines from the design team, but for now, it's less important?

kbventures reviewed (commented):
Great work @manvinderjit 

Small ask.  For future branches lets create a draft sooner during the development project so we are able to coordinate more efficiently have have a pulse on progression.

kbventures commented:
@manvinderjit @madcampos 

Should we introduce PascalCase for the components?

https://www.sufle.io/blog/naming-conventions-in-react

kbventures commented:
@manvinderjit nit picking but eventually we probably want Inner Components (Reusable UI Elements) for Team & Profile

manvinderjit commented:
> @manvinderjit @madcampos
> 
> Should we introduce PascalCase for the components?
> 
> https://www.sufle.io/blog/naming-conventions-in-react

I am already following PascalCase for components, unless I might have missed it somewhere. Functions and variables are camel case.

manvinderjit commented:
> @manvinderjit nit picking but eventually we probably want Inner Components (Reusable UI Elements) for Team & Profile

Kindly elaborate on the elements parts. Do you mean the card for each team member and profile, if yes then these are already reusable components -> TeamMemberCard & ProfileCard.

Otherwise, lemme know.

manvinderjit commented:
> Please test the page for mobile using the dev tools to check for potential issues with content breaking.
> 
> ![localhost_3000_pages_profile__pid=2](https://private-user-images.githubusercontent.com/609904/407633613-2dcbda94-1947-4806-a751-91d797594bde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxODIxODgsIm5iZiI6MTczODE4MTg4OCwicGF0aCI6Ii82MDk5MDQvNDA3NjMzNjEzLTJkY2JkYTk0LTE5NDctNDgwNi1hNzUxLTkxZDc5NzU5NGJkZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyOVQyMDE4MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGU5MDc1NDI4YmQ5MjI0MzIwNDExZDFlZWU5YjVhOWM1ZDFmZDQ0Y2UwOTQwZGVjMjI1MjE4YjVlOTdmMDI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nPbGMRg4EFIy61e8Et2-e5c9MsTcxNnhKe0DQbRx6jE)

OOPS! I might have not checked below 600px width. Will rectify it immediately!

manvinderjit committed:
fix: fix mobile content breaking

Fix mobile content breaking under 600px.
Update message pattern in ProfileCard component.

kbventures commented:
It was simply an mvp to compare deployment solutions.

manvinderjit commented:
> Great work @manvinderjit
> 
> Small ask. For future branches lets create a draft sooner during the development project so we are able to coordinate more efficiently have have a pulse on progression.

@kbventures Will do.

kbventures commented:
@manvinderjit component inner component and camelcase are fine. Ignore that.

madcampos commented:
@kbventures
> Should we introduce PascalCase for the components?

Yes, react component names should be in PascalCase, always, that helps to differentiate from native html elements and web components when referencing them in jsx.

Fun fact: this is also a recommendation in vue for the same reason, it helps differentiate framework components from other things.

> > @manvinderjit @madcampos
> > Should we introduce PascalCase for the components?
> > https://www.sufle.io/blog/naming-conventions-in-react
> 
> I am already following PascalCase for components, unless I might have missed it somewhere. Functions and variables are camel case. I can update the contribution guides if required for the same.
> 
> For the pages, I am using small case because that is how @madcampos initialized them. Not sure on the nomenclature for them.

Ops... That was an oversight on my side

kbventures committed:
Merge branch 'dev' into feat/page-profile
